huaweicloud/services/live/resource_huaweicloud_live_domain.go

```diff
@@ -228,175 +279,176 @@ func resourceDomainUpdate(ctx context.Context, d *schema.ResourceData, meta inte
-	} else {
-		reqStatus = model.GetLiveDomainModifyReqStatusEnum().ON
-		respStatus = model.GetDecoupledLiveDomainInfoStatusEnum().ON
+func buildUpdateaDomainStatusBodyParams(cfg *config.Config, d *schema.ResourceData,
+	status string) map[string]interface{} {
+	params := map[string]interface{}{
+		"domain":                d.Get("name"),
+		"status":                status,
+		"enterprise_project_id": utils.ValueIgnoreEmpty(cfg.GetEnterpriseProjectID(d)),
 	}
 
-	_, err := client.UpdateDomain(&model.UpdateDomainRequest{
-		Body: &model.LiveDomainModifyReq{
-			Domain:              d.Get("name").(string),
-			Status:              &reqStatus,
-			EnterpriseProjectId: utils.StringIgnoreEmpty(cfg.GetEnterpriseProjectID(d)),
-		},
-	})
+	return params
+}
 
+func updateIPv6Switch(client *golangsdk.ServiceClient, d *schema.ResourceData, domainName string) error {
+	httpUrl := "v1/{project_id}/domain/ipv6-switch"
+	requestPath := client.Endpoint + httpUrl
+	requestPath = strings.ReplaceAll(requestPath, "{project_id}", client.ProjectID)
+	requestOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		JSONBody:         buildUpdateIPv6SwitchBodyParams(d, domainName),
+	}
+
+	_, err := client.Request("PUT", requestPath, &requestOpt)
 	if err != nil {
-		return fmt.Errorf("error updating Live domain: %s", err)
+		return fmt.Errorf("error enabling or disabling the IPv6: %s", err)
 	}
 
-	return waitingForDomainStatus(ctx, client, d.Id(), respStatus, d.Timeout(schema.TimeoutUpdate))
+	return nil
 }
 
-func updateIPv6Switch(client *livev1.LiveClient, d *schema.ResourceData) error {
-	switchRequest := model.UpdateDomainIp6SwitchRequest{
-		Body: &model.DomainIpv6SwitchReq{
-			Domain: d.Get("name").(string),
-			IsIpv6: utils.Bool(d.Get("is_ipv6").(bool)),
-		},
+func buildUpdateIPv6SwitchBodyParams(d *schema.ResourceData, domainName string) map[string]interface{} {
+	params := map[string]interface{}{
+		"domain":  domainName,
+		"is_ipv6": utils.ValueIgnoreEmpty(d.Get("is_ipv6")),
 	}
 
-	_, err := client.UpdateDomainIp6Switch(&switchRequest)
+	return params
+}
+
+func waitingForDomainStateCompleted(ctx context.Context, client *golangsdk.ServiceClient, t time.Duration, domainName string) error {
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{"PENDING"},
+		Target:       []string{"COMPLETED"},
+		Refresh:      waitDomainStatusRefreshFunc(client, domainName),
+		Timeout:      t,
+		Delay:        10 * time.Second,
+		PollInterval: 10 * time.Second,
+	}
+
+	_, err := stateConf.WaitForStateContext(ctx)
 	return err
 }
+
+func waitDomainStatusRefreshFunc(client *golangsdk.ServiceClient, domainName string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		respBody, err := GetDomain(client, domainName)
+		if err != nil {
+			return nil, "ERROR", err
+		}
+
+		status := utils.PathSearch("status", respBody, "").(string)
+
+		if utils.StrSliceContains([]string{"on", "off"}, status) {
```

Comment:
The code logic is not rigorous, it is recommended to judge the expected value separately.